system-x/services/fhir/src/main/java/software/tnb/fhir/resource/openshift/OpenshiftFhir.java

```diff
@@ -0,0 +1,123 @@
+        OpenShiftWaiters.get(OpenshiftClient.get(), () -> false)
+            .areNoPodsPresent(OpenshiftConfiguration.openshiftDeploymentLabel(), name()).timeout(120_000).waitFor();
+    }
+
+    @Override
+    public void create() {
+        OpenshiftClient.get().deploymentConfigs().createOrReplace(new DeploymentConfigBuilder()
+            .withNewMetadata()
+                .withName(name())
+                .addToLabels(OpenshiftConfiguration.openshiftDeploymentLabel(), name())
+            .endMetadata()
+                .editOrNewSpec()
+                    .addToSelector(OpenshiftConfiguration.openshiftDeploymentLabel(), name())
+                    .withReplicas(1)
+                    .editOrNewTemplate()
+                        .editOrNewMetadata()
+                            .addToLabels(OpenshiftConfiguration.openshiftDeploymentLabel(), name())
+                        .endMetadata()
+                        .editOrNewSpec()
+                            .addNewContainer()
+                                .withName(name())
+                                .withImage(fhirImage())
+                                .addNewPort()
+                                    .withContainerPort(PORT)
+                                    .withName(name())
+                                .endPort()
+                                .addAllToEnv(containerEnvironment().entrySet().stream().map(e -> new EnvVar(e.getKey(), e.getValue(), null))
+                                    .collect(Collectors.toList()))
+                            .endContainer()
+                        .endSpec()
+                    .endTemplate()
+                    .addNewTrigger()
+                        .withType("ConfigChange")
+                    .endTrigger()
+                .endSpec()
+            .build());
+
+        OpenshiftClient.get().services().createOrReplace(new ServiceBuilder()
+            .editOrNewMetadata()
+                .withName(name())
+                .addToLabels(OpenshiftConfiguration.openshiftDeploymentLabel(), name())
+            .endMetadata()
+            .editOrNewSpec()
+                .addToSelector(OpenshiftConfiguration.openshiftDeploymentLabel(), name())
+                .addNewPort()
+                    .withName(name())
+                    .withProtocol("TCP")
+                    .withPort(PORT)
+                    .withTargetPort(new IntOrString(PORT))
+                .endPort()
+            .endSpec()
+        .build());
+    }
+
+    @Override
+    public boolean isReady() {
+        List<Pod> list = OpenshiftClient.get().getLabeledPods(OpenshiftConfiguration.openshiftDeploymentLabel(), name());
+        return ResourceFunctions.areExactlyNPodsReady(1).apply(list);
+    }
+
+    @Override
+    public boolean isDeployed() {
+        return OpenshiftClient.get().getLabeledPods(OpenshiftConfiguration.openshiftDeploymentLabel(), name()).size() > 0 && isReady();
```

Comment:
`isDeployed` shouldn't use `isReady`, those should be two separate things

Reply:
then you should ask people to rework FTP, Mongo and Http on OCP. This is just copy pasta.

Reply:
I hope just removing isReady is fine